woocommerce/woocommerce
# Add filters to Variation table

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #39890



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure feature `New product editor` is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
2. Make sure feature `product-variation-management` is enabled under `Features` tab from `/wp-admin/tools.php?page=woocommerce-admin-test-helper` (WooCommerce Beta Tester plugin) is required
3. Go to `/wp-admin/admin.php?page=wc-admin&path=/add-product&tab=variations` and add some attributes to generate variations
4. Wait until variations get generated
5. For each `Variation options` a new filter on top of the `Variations` table should be shown. 
<img width="695" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/21693b97-0bbe-47c3-a921-e9b2653e1aca">

6. Each filter should be shown as a dropdown list and it should list all the options/terms that belongs to the selected attribute 
<img width="695" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/7b67e6b0-cf8b-4c83-8e55-2d60ceefc21d">

7. The list should support multi-selection so the users can select 0 or many terms to filter by 
<img width="699" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/08f2370e-24a7-4f15-a78b-8a448ca0f708">

8. The dropdown list should have a footer with Reset button which will clear all selected terms when click it and should be disabled if there are no selected terms 
<img width="689" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/42b21ba4-0125-4dba-85b0-8693269b6535">

9. The dropdown list should have a footer with Filter button which will filter all the variations using the selected terms 
<img width="706" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/f0d62359-a311-463f-9e6f-e3d5a0a1ae1a">




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
